src/ast_utils.h

```diff
@@ -457,431 +282,18 @@ struct ExpressionAnalyzer {
-            PUSH(CallImport, operands[i]);
-          }
-          break;
-        }
-        case Expression::Id::CallIndirectId: {
-          PUSH(CallIndirect, target);
-          CHECK(CallIndirect, fullType);
-          CHECK(CallIndirect, operands.size());
-          for (Index i = 0; i < left->cast<CallIndirect>()->operands.size(); i++) {
-            PUSH(CallIndirect, operands[i]);
-          }
-          break;
-        }
-        case Expression::Id::GetLocalId: {
-          CHECK(GetLocal, index);
-          break;
-        }
-        case Expression::Id::SetLocalId: {
-          CHECK(SetLocal, index);
-          CHECK(SetLocal, type); // for tee/set
-          PUSH(SetLocal, value);
-          break;
-        }
-        case Expression::Id::GetGlobalId: {
-          CHECK(GetGlobal, name);
-          break;
-        }
-        case Expression::Id::SetGlobalId: {
-          CHECK(SetGlobal, name);
-          PUSH(SetGlobal, value);
-          break;
-        }
-        case Expression::Id::LoadId: {
-          CHECK(Load, bytes);
-          CHECK(Load, signed_);
-          CHECK(Load, offset);
-          CHECK(Load, align);
-          PUSH(Load, ptr);
-          break;
-        }
-        case Expression::Id::StoreId: {
-          CHECK(Store, bytes);
-          CHECK(Store, offset);
-          CHECK(Store, align);
-          CHECK(Store, valueType);
-          PUSH(Store, ptr);
-          PUSH(Store, value);
-          break;
-        }
-        case Expression::Id::ConstId: {
-          CHECK(Const, value);
-          break;
-        }
-        case Expression::Id::UnaryId: {
-          CHECK(Unary, op);
-          PUSH(Unary, value);
-          break;
-        }
-        case Expression::Id::BinaryId: {
-          CHECK(Binary, op);
-          PUSH(Binary, left);
-          PUSH(Binary, right);
-          break;
-        }
-        case Expression::Id::SelectId: {
-          PUSH(Select, ifTrue);
-          PUSH(Select, ifFalse);
-          PUSH(Select, condition);
-          break;
-        }
-        case Expression::Id::DropId: {
-          PUSH(Drop, value);
-          break;
-        }
-        case Expression::Id::ReturnId: {
-          PUSH(Return, value);
-          break;
-        }
-        case Expression::Id::HostId: {
-          CHECK(Host, op);
-          CHECK(Host, nameOperand);
-          CHECK(Host, operands.size());
-          for (Index i = 0; i < left->cast<Host>()->operands.size(); i++) {
-            PUSH(Host, operands[i]);
-          }
-          break;
-        }
-        case Expression::Id::NopId: {
-          break;
-        }
-        case Expression::Id::UnreachableId: {
-          break;
-        }
-        default: WASM_UNREACHABLE();
-      }
-      #undef CHECK
-      #undef PUSH
-    }
-    if (leftStack.size() > 0 || rightStack.size() > 0) return false;
-    return true;
-  }
+  using ExprComparer = std::function<bool(Expression*, Expression*)>;
```

Comment:
It's basically a nicer way of spelling `typedef` in C++11: http://en.cppreference.com/w/cpp/language/type_alias